redis/src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -368,40 +473,71 @@ trait SortedSets {
    * @param limit The optional LIMIT argument can be used to only get a range of the matching elements. A negative count returns all elements from the offset
    * @return Chunk of elements in the specified score range
    */
-  final def zRevRangeByLex(
-    key: String,
+  final def zRevRangeByLex[K: Schema, M: Schema](
+    key: K,
     lexRange: LexRange,
     limit: Option[Limit] = None
-  ): ZIO[RedisExecutor, RedisError, Chunk[String]] = {
+  ): ZIO[RedisExecutor, RedisError, Chunk[M]] = {
     val command = RedisCommand(
       ZRevRangeByLex,
-      Tuple3(StringInput, LexRangeInput, OptionalInput(LimitInput)),
-      ChunkOutput(MultiStringOutput)
+      Tuple4(ArbitraryInput[K](), ArbitraryInput[String](), ArbitraryInput[String](), OptionalInput(LimitInput)),
+      ChunkOutput(ArbitraryOutput[M]())
     )
-    command.run((key, lexRange, limit))
+    command.run((key, lexRange.max.stringify, lexRange.min.stringify, limit))
   }
 
   /**
    * Return a range of members in a sorted set, by score, with scores ordered from high to low.
    *
    * @param key Key of a sorted set
    * @param scoreRange ScoreRange that must be satisfied
-   * @param withScores The optional WITHSCORES argument makes the command return both the element and its score, instead of the element alone
    * @param limit The optional LIMIT argument can be used to only get a range of the matching elements. A negative count returns all elements from the offset
-   * @return Chunk of elements in the specified range (optionally with their scores)
+   * @return Chunk of elements in the specified range
    */
-  final def zRevRangeByScore(
-    key: String,
+  final def zRevRangeByScore[K: Schema, M: Schema](
+    key: K,
     scoreRange: ScoreRange,
-    withScores: Option[WithScores] = None,
     limit: Option[Limit] = None
-  ): ZIO[RedisExecutor, RedisError, Chunk[String]] = {
+  ): ZIO[RedisExecutor, RedisError, Chunk[M]] = {
     val command = RedisCommand(
       ZRevRangeByScore,
-      Tuple4(StringInput, ScoreRangeInput, OptionalInput(WithScoresInput), OptionalInput(LimitInput)),
-      ChunkOutput(MultiStringOutput)
+      Tuple4(
+        ArbitraryInput[K](),
+        ArbitraryInput[String](),
+        ArbitraryInput[String](),
+        OptionalInput(LimitInput)
+      ),
+      ChunkOutput(ArbitraryOutput[M]())
+    )
+    command.run((key, scoreRange.max.stringify, scoreRange.min.stringify, limit))
```

Comment:
I wonder if it would be better to also have a ScoreRangeInput like we have for example for MemberScore